src/core/lib/channel/channelz_registry.cc

```diff
@@ -59,6 +66,10 @@ void ChannelzRegistry::InternalRegister(BaseNode* node) {
 }
 
 void ChannelzRegistry::InternalUnregister(intptr_t uuid) {
+  if (GPR_UNLIKELY(!g_channelz_registry_running)) {
+    gpr_log(GPR_ERROR, "InternalUnregister invoked after Shutdown");
+    return;
```

Comment:
Actually, I take it back -- this is not actually a memory leak, because the registry is not holding a ref to the node.  However, it is still a hacky work-around.

I think we need to solve the underlying problem, not work around the symptom.